nemoguardrails/embeddings/basic.py

```
@@ -187,74 +191,105 @@ async def build(self):
+                    f"Embedding model returned {len(embeddings)} embeddings for {len(batch_ids)} inputs."
+                )
+                for req_id in batch_ids[len(embeddings) :]:
+                    self._req_errors[req_id] = shortage_exc
+            for i in range(len(embeddings)):
+                self._req_results[batch_ids[i]] = embeddings[i]
+        except asyncio.CancelledError as exc:
+            for req_id in batch_ids:
+                if req_id not in self._req_results and req_id not in self._req_errors:
+                    self._req_errors[req_id] = exc
+            raise
+        except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)